python/lsst/ts/externalscripts/base_make_bias.py

```diff
@@ -0,0 +1,227 @@
+            externalscripts/maintel/make_comcam_bias.py
+        title: BaseMakeBias v1
+        description: Configuration for BaseMakeBias.
+        type: object
+        properties:
+            n_bias:
+                type: integer
+                default: 1
+                description: number of biases to take
+
+            detectors:
+                type: string
+                items:
+                    type: integer
+                    minItems: 1
+                default: (0)
+                descriptor: Detector IDs
+
+            input_collections:
+                type: string
+                descriptor: Input collections to pass to the bias pipetask.
+
+            calib_dir:
+                type: string
+                descriptor: path to the calib directory for the bias when certifying it.
+
+            repo:
+                type: string
+                descriptor: Butler repository. /repo/main is the default at NCSA;\
+                    it migth be different at the summit.
+                default: /repo/main
+        additionalProperties: false
+        required: [input_collections, calib_dir, repo]
+        """
+        return yaml.safe_load(schema)
+
+    async def configure(self, config):
+        """Configure the script.
+
+        Parameters
+        ----------
+        config: `types.SimpleNamespace`
+            Configuration data. See `get_schema` for information about data
+            structure.
+        """
+        # Log information about the configuration
+
+        self.log.debug(
+            f"n_bias: {config.n_bias}, detectors: {config.detectors}, "
+            f"instrument: {self.instrument_name} "
+        )
+
+        self.config = config
+
+    def set_metadata(self, metadata):
+        """Set estimated duration of the script.
+        """
+        metadata.duration = self.config.n_bias*(self.camera.read_out_time + self.estimated_process_time)
+
+    async def arun(self, checkpoint=False):
+
+        # Take config.n_biases biases, and return a list of IDs
+        if checkpoint:
+            await self.checkpoint(f"Taking {self.config.n_bias} biases")
+
+        self.image_in_oods.flush()
+        exposures = tuple(await self.camera.take_bias(self.config.n_bias))
+
+        if checkpoint:
+            # Bias IDs
+            await self.checkpoint(f"Biases taken: {exposures}")
+
+        # did the images get archived and are they available to the butler?
+        n_detectors = len(tuple(map(int, self.config.detectors[1:-1].split(','))))
+        # exps are of the form, e.g., 2021070800019
+        exposure_set = set()
+        for exp in exposures:
+            obs_day = f"{exp}"[:8]
+            temp = int(f"{exp}"[8:])
+            # See example of ack.obsid below
+            seq_num = f"{temp:06}"
+            obs_id = obs_day + "_" + seq_num
+            exposure_set.add(obs_id)
+
+        images_remaining = len(exposure_set)*n_detectors
+        max_counter = 1000
```

Comment:
consider making `max_counter` either a class attribute (`self.max_counter=1000` in `__init__`) or making it a configuration parameter....